overlord/snapstate/snapmgr.go

```diff
@@ -1084,6 +1086,15 @@ func MockEnsuredMountsUpdated(m *SnapManager, ensured bool) (restore func()) {
 	}
 }
 
+func MockEnsuredDesktopFilesUpdated(m *SnapManager, ensured bool) (restore func()) {
```

Comment:
do we need to call this from any of the places currently calling MockEnsuredMountsUpdated ? if some snaps with desktop files get involved in those tests we might want to do that already now? otherwise this could just live in export_test.go

Reply:
Yep, I think it can be moved to export_test.go.

I didn't notice problems with any other tests. I suspect most of the other calls to MockEnsuredMountsUpdated were to prevent the test from trying to talk to systemd. ensureDesktopFilesUpdated may call a subprocess in some cases, but that's only going to affect files in the test's private temporary directory.